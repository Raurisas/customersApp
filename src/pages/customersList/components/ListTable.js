import React from 'react';
import ListRow from './ListRow';

const ListTable = ({ items, handleRowClick, handleRemoveClick, handleEditClick }) => {
  return (
    <div className="list-table">
      {items.map(item => (
        <ListRow
          item={item}
          key={item.id}
          handleRowClick={handleRowClick}
          handleRemoveClick={handleRemoveClick}
          handleEditClick={handleEditClick}
        />
      ))}
    </div>
  );
};

export default ListTable;
