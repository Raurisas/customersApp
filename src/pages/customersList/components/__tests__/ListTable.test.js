import React from 'react';
import { shallow } from 'enzyme';
import ListTable from '../ListTable';

it('should match snapshot', () => {
  const items = [
    {
      id: 1,
      email: 'aurimas993@gmail.com',
      metadata: {
        fullName: 'Aurimas Jackevicius',
        city: 'Vilnius',
        street: 'Ulvydo',
        houseNumber: '7',
        zipCode: '08250'
      }
    },
    {
      id: 2,
      email: 'aurimas993@gmail.com',
      metadata: {
        fullName: 'Aurimas Jackevicius',
        city: 'Vilnius',
        street: 'Ulvydo',
        houseNumber: '7',
        zipCode: '08250'
      }
    }
  ];

  const wrapper = shallow(
    <ListTable items={items} handleRowClick={() => {}} handleRemoveClick={() => {}} handleEditClick={() => {}} />
  );
  expect(wrapper).toMatchSnapshot();
});
