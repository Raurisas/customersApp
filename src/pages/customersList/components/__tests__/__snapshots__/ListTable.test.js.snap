// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="list-table"
>
  <ListRow
    handleEditClick={[Function]}
    handleRemoveClick={[Function]}
    handleRowClick={[Function]}
    item={
      Object {
        "email": "aurimas993@gmail.com",
        "id": 1,
        "metadata": Object {
          "city": "Vilnius",
          "fullName": "Aurimas Jackevicius",
          "houseNumber": "7",
          "street": "Ulvydo",
          "zipCode": "08250",
        },
      }
    }
    key="1"
  />
  <ListRow
    handleEditClick={[Function]}
    handleRemoveClick={[Function]}
    handleRowClick={[Function]}
    item={
      Object {
        "email": "aurimas993@gmail.com",
        "id": 2,
        "metadata": Object {
          "city": "Vilnius",
          "fullName": "Aurimas Jackevicius",
          "houseNumber": "7",
          "street": "Ulvydo",
          "zipCode": "08250",
        },
      }
    }
    key="2"
  />
</div>
`;
